pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -234,7 +252,7 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 		}
 	}
 
-	err = service.UpdateKubernetesInlineComponentsOwnerReferences(a.Client.GetKubeClient(), k8sComponents, ownerReference)
+	err = service.UpdateServicesWithOwnerReferences(a.Client.GetKubeClient(), k8sComponents, ownerReference)
```

Comment:
As discussed during the daily meeting, you should check at this point that the owner reference is ok (it could be wrong if the deployment was not created), and rollback the services creation if it is wrong.

Reply:
I have created a separate issue to rollback both services and storage in such a scenario https://github.com/openshift/odo/issues/4953